man/systemd-rc-local-generator.xml

```diff
@@ -19,28 +19,40 @@
 
   <refnamediv>
     <refname>systemd-rc-local-generator</refname>
-    <refpurpose>Compatibility generator for starting <filename>&RC_LOCAL_PATH;</filename> during boot</refpurpose>
+    <refname>rc-local.service</refname>
+    <refpurpose>Compatibility generator and service to start <filename>&RC_LOCAL_PATH;</filename> during boot</refpurpose>
   </refnamediv>
 
   <refsynopsisdiv>
     <para><filename>/usr/lib/systemd/system-generators/systemd-rc-local-generator</filename></para>
+    <para><filename>rc-local.service</filename></para>
   </refsynopsisdiv>
 
   <refsect1>
     <title>Description</title>
 
-    <para><filename>systemd-rc-local-generator</filename> is a generator that checks whether
-    <filename>&RC_LOCAL_PATH;</filename> exists and is executable, and if it is pulls the
+    <para><command>systemd-rc-local-generator</command> is a generator that checks whether
+    <filename>&RC_LOCAL_PATH;</filename> exists and is executable, and if it is, pulls the
     <filename>rc-local.service</filename> unit into the boot process. This unit is responsible for running
-    this script during late boot. Note that the script will be run with slightly different semantics than the
-    original System V version, which was run "last" in the boot process, which is a concept that does not
-    translate to systemd. The script is run after <filename>network.target</filename>, but in parallel with
-    most other regular system services.</para>
+    this script during late boot. The script is run after <filename>network.target</filename>, but in
+    parallel with most other regular system services.</para>
+
+    <para>Note that <filename>rc-local.service</filename> runs with slightly different semantics than the
+    original System V version, which was executed "last" in the boot process, which is a concept that does
+    not translate to systemd.</para>
+
+    <para>Also note that <filename>rc-local.service</filename> is ordered after
+    <filename>network.target</filename>, which does not mean that the network is functional, see
+    <citerefentry><refentrytitle>systemd.special</refentrytitle><manvolnum>7</manvolnum></citerefentry>.
+    If the script requires a configured network connection, it may be desirable to pull in and order it after
+    <filename>network-online.target</filename> with:</para>
+
+    <programlisting>systemctl add-wants network-online.target rc-local.service</programlisting>
```

Comment:
Isn't this the wrong way around?
rc-local.service should have wants/after network-online.target, not network-online.target wanting rc-local.service?

Reply:
`systemctl add-wants` doesn't add add a `After=` ordering, only a `Wants=` depdendency, right?

Reply:
Yikes, thanks for catching that!

> `systemctl add-wants` doesn't add add a `After=` ordering, only a `Wants=` depdendency, right?

A symlink in  .wants/.requires implies also the ordering dependency After.

Reply:
> A symlink in .wants/.requires implies also the ordering dependency After.

Hm, I must be doing something wrong then:
```
# systemctl show -p Wants -p After rc-local.service
Wants=
After=basic.target sysinit.target system.slice systemd-journald.socket network.target

# systemctl add-wants rc-local.service network-online.target
Created symlink /etc/systemd/system/rc-local.service.wants/network-online.target → /lib/systemd/system/network-online.target.

# systemctl show -p Wants -p After rc-local.service
Wants=network-online.target
After=basic.target sysinit.target network.target systemd-journald.socket system.slice
```

Reply:
Hmm. Indeed. I thought that this is the case, but the man page says only:
> target units will complement all configured dependencies of type Wants= or Requires= with dependencies of type After=

It seems @poettering thought so too:
> But if we ask people that, then we could as well ask them to instead do:
>
> mkdir -p /etc/systemd/system/rc-local.service.wants/
> ln -s /usr/lib/systemd/system/network-online.target /etc/systemd/system/rc-local.service.wants/
> 
> Which actually will do both: add Wants= + After=, and thus be more comprehensive.

Duh.

Reply:
OK, I updated the man page to suggest a drop-in. It's not very nice, but I don't see a better approach.